src/main/java/BookController.java

```diff
@@ -39,7 +42,7 @@ public void delete(@PathVariable Long id) {
     }
 
     @PutMapping("/{id}")
-    public Book updateBook(@RequestBody Book book, @PathVariable Long id) {
+    public Book updateBook(@RequestBody Book book, @PathVariable Long id) throws BookIdMismatchException {
```

Comment:
Because we're explicitly throwing an exception within the method, that needs to be part of method signature